app/helpers/project_nav_helper.rb

```diff
@@ -0,0 +1,32 @@
+module ProjectNavHelper
+
+  def next_project_id project
+    return if is_last_project? project
+
+    all_keys = chapter_projects_keys(project)
+    index_next_project = all_keys.index(project.id) + 1
+    all_keys[index_next_project]
+  end
+
+  def prev_project_id project
+    return if is_first_project? project
+
+    all_keys = chapter_projects_keys(project)
+    index_prev_project = all_keys.index(project.id) - 1
+    all_keys[index_prev_project]
+  end
+
+  private
+
+  def chapter_projects_keys(project)
+    project.chapter.projects.order('funded_on DESC').pluck(:id)
```

Comment:
Also, this should be `project.chapter.winning_projects` - since this is on a public project page, it should only show "funded" projects (i.e. ones that have a `funded_on` date set). The tests should validate this as well.

Reply:
> Is there a reason this isn't memoized?

In general, I try to follow a functional approach, so that methods do not share state. Esp. helpers, since they are stateless modules per se, and their state object (`self`) might be not what we expect.

The proper way IMO is to use either cache or ADT. Both seemed like an overkill to me, the queries are simple and the DB is small. (Also, I've checked other helpers, they seem to not care about DB queries much as well.)

If you think caching this computation as an instance variable is fine, I can do that.

Reply:
> Also, this should be project.chapter.winning_projects

Thanks, this wasn't clear from the code :+1: